src/averageAges.js

```diff
@@ -36,8 +52,17 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomenPeople = people.filter(el => {
+    return el.sex !== SEX_FEMALE
+      ? false
+      : withChildren === true
+        ? people.some(({ mother }) => mother === el.name)
+        : true;
+  });
```

Comment:
Don't compare with undefined/null or empty string. Use the logical not to check the falsy values.